Update Default Homepage Cards to Reflect the New Design

Eswaraiahsapram opened the pull request:
## Default homepage update to new design

This PR contains following changes:

- Updated default Homepage cards
- Updated dev default cards
- Updated with new cards. Onboarding section, Entity section and Template section cards

## Screenshots

### Old cards
<img width="1690" alt="image" src="https://github.com/user-attachments/assets/e2851cf4-0b0c-448c-a473-04878a7c14d3" />
<img width="1695" alt="image" src="https://github.com/user-attachments/assets/3a8f986f-5fed-4a39-9a65-2143ec7be27f" />


### New cards
<img width="1669" alt="Screenshot 2025-06-19 at 11 54 07 AM" src="https://github.com/user-attachments/assets/0f792922-238d-459d-aae7-f99ba6a3c0c9" />

![image](https://github.com/user-attachments/assets/1978a34c-baf1-4e7f-b8ed-977f2620fd1a)

### Together
![image](https://github.com/user-attachments/assets/38be005b-2b91-4ffa-9cfe-1d6764bea8b4)
<img width="1672" alt="image" src="https://github.com/user-attachments/assets/f4387497-b2cd-4b2e-b8fd-1984bed410b5" />



#### :heavy_check_mark: Checklist

<!--- Please include the following in your Pull Request when applicable: -->

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/redhat-developer/rhdh-plugins/blob/main/CONTRIBUTING.md#creating-changesets))
- [x] Added or Updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)

rohitkrai03 commented:
@Eswaraiahsapram Can you add some screenshots of how this new default homepage looks like now with the new cards and old cards together?

Eswaraiahsapram commented:
> It seems like the cards are not responsive for smaller screen layout.

@rohitkrai03 Initially, I used `MUI <Grid> with xs, md, and lg` breakpoints and also attempted `media queries` for responsive layout. However, since the layout is container-based (not screen-based) and I don't have control over the parent container's width (which can vary independently of screen size), media queries and Grid breakpoints didn't provide the expected layout consistency.

So, I switched to a `flex-wrap` layout using `ResizeObserver` to track the container's actual width. Based on this width, I calculated the cardWidth dynamically using `calc()` and adjusted for gap to get accurate spacing.

![Screenshot 2025-06-23 at 11 38 33 AM](https://github.com/user-attachments/assets/8a41b478-b461-4a94-b723-de0d35e371c8)

![Screenshot 2025-06-23 at 11 38 49 AM](https://github.com/user-attachments/assets/84e117b0-0f70-4c4a-b023-78f4a988e6a6)

ciiay reviewed (commented):
Hi @Eswaraiahsapram , thanks for the PR. I tested both in the dev app and in RHDH. There are some differences.
Note that the text size in tags looks larger or too closed to tag border in RHDH.

Dev app
![image](https://github.com/user-attachments/assets/3b2c3ae5-af39-4d4b-a0e6-707b4161280b)

RHDH
![image](https://github.com/user-attachments/assets/8f7513e0-2437-482e-a033-a6c43ef65405)

Eswaraiahsapram commented:
> Hi @Eswaraiahsapram , thanks for the PR. I tested both in the dev app and in RHDH. There are some differences. Note that the text size in tags looks larger or too closed to tag border in RHDH.
> 
Hi @ciiay , thanks a lot for the feedback.

The difference in tag styles is likely due to the missing `@red-hat-developer-hub/backstage-plugin-theme` package in the dev app. This theme is applied in RHDH, which is why the text size and spacing appear different.

ciiay commented:
>The difference in tag styles is likely due to the missing @red-hat-developer-hub/backstage-plugin-theme package in the dev app. This theme is applied in RHDH, which is why the text size and spacing appear different.

Ok, then we might need to update the text padding or enlarge the tag size in theme because according to the [figma design](https://www.figma.com/design/6ORS1mhUaxeyTBjMmE95Rn/Home-page-redesign?node-id=787-203&t=fz5GDeUdcVqYoEQ0-0), the text padding look wider than how it looks in RHDH. We can address this issue in a different story.

@ShiranHi what do you think?

ShiranHi commented:
> > The difference in tag styles is likely due to the missing @red-hat-developer-hub/backstage-plugin-theme package in the dev app. This theme is applied in RHDH, which is why the text size and spacing appear different.
> 
> Ok, then we might need to update the text padding or enlarge the tag size in theme because according to the [figma design](https://www.figma.com/design/6ORS1mhUaxeyTBjMmE95Rn/Home-page-redesign?node-id=787-203&t=fz5GDeUdcVqYoEQ0-0), the text padding look wider than how it looks in RHDH. We can address this issue in a different story.
> 
> @ShiranHi what do you think?

The tags in [this screenshot](https://private-user-images.githubusercontent.com/26255262/458544565-3b2c3ae5-af39-4d4b-a0e6-707b4161280b.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTA5MTQ2MzcsIm5iZiI6MTc1MDkxNDMzNywicGF0aCI6Ii8yNjI1NTI2Mi80NTg1NDQ1NjUtM2IyYzNhZTUtYWYzOS00ZDRiLWEwZTYtNzA3YjQxNjEyODBiLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA2MjYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNjI2VDA1MDUzN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWI2ZjYwNDk1NmIzZjU0ZDAzMWJhNDc1Y2U0YTU5M2NlODU2NTk0MjlmNjY0YmZkZjcxMDQwZWJhYWNiNDNmNGEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.5oU1F26sjNneMWCJ0pptB7-wcz6aYa3CZ5qSkymWFDo) appear acceptable to me. Could you please provide another screenshot so I can better understand the issue you're referring to?

rohitkrai03 commented on workspaces/homepage/plugins/dynamic-home-page/src/components/EntitySection/EntitySection.tsx:
```diff
@@ -60,6 +60,27 @@ export const EntitySection = () => {
+  const { ref, width = 1 } = useResizeObserver();
+  const gap = 16;
+
+  useEffect(() => {
+    if (width >= 900 && !isRemoveFirstCard && !profileLoading) {
+      setImgCardWidth(`calc(42%)`);
+      setCardWidth(`calc((56% - ${gap}px) / 2)`);
+    } else if (width >= 900) {
+      setCardWidth(`calc((100% - ${gap * 3}px) / 4)`);
+    } else if (width > 600) {
+      setImgCardWidth(`calc((100% - ${gap * 2}px) / 2)`);
+      setCardWidth(`calc((100% - ${gap * 2}px) / 2)`);
+    } else if (width <= 600) {
+      setImgCardWidth(`calc(100%)`);
+      setCardWidth(`calc((100%))`);
+    }
+  }, [profileLoading, isRemoveFirstCard, width, gap]);
```
@Eswaraiahsapram I am still not sure if using `ResizeObserver` is the right thing here. We should not be needing this kind of computation for implementing just a simple responsive layout.

Eswaraiahsapram replied:
You're right, @rohitkrai03. I also feel `ResizeObserver` shouldn't be needed for a simple responsive layout. The issue only appears in standalone mode. Mostly due to how the page is structured there. I discussed this with @karthikjeeyar as well, and he suggested a few approaches, which I'm currently trying out.
